client/src/actions/monument.js

```diff
@@ -5,66 +5,25 @@ import { FETCH_MONUMENT_PENDING, FETCH_MONUMENT_ERROR, FETCH_MONUMENT_SUCCESS, F
 import * as QueryString from 'query-string';
 import { get } from '../utils/api-util';
 import { addError } from './errors';
-
-function fetchMonumentPending() {
-    return {
-        type: FETCH_MONUMENT_PENDING
-    };
-}
-
-function fetchMonumentSuccess(monument) {
-    return {
-        type: FETCH_MONUMENT_SUCCESS,
-        payload: monument
-    };
-}
-
-function fetchMonumentError(error) {
-    return {
-        type: FETCH_MONUMENT_ERROR,
-        error: error
-    };
-}
-
-function fetchNearbyMonumentsPending() {
-    return {
-        type: FETCH_NEARBY_MONUMENTS_PENDING
-    };
-}
-
-function fetchNearbyMonumentsSuccess(monuments) {
-    return {
-        type: FETCH_NEARBY_MONUMENTS_SUCCESS,
-        payload: monuments
-    };
-}
-
-function fetchNearbyMonumentsError(error) {
-    return {
-        type: FETCH_NEARBY_MONUMENTS_ERROR,
-        error: error
-    };
-}
-
-function fetchRelatedMonumentsPending() {
-    return {
-        type: FETCH_RELATED_MONUMENTS_PENDING
-    };
-}
-
-function fetchRelatedMonumentsSuccess(monuments) {
-    return {
-        type: FETCH_RELATED_MONUMENTS_SUCCESS,
-        payload: monuments
-    };
-}
-
-function fetchRelatedMonumentsError(error) {
-    return {
-        type: FETCH_RELATED_MONUMENTS_ERROR,
-        error: error
-    };
-}
+import { pending, success, error } from '../utils/action-util';
+
+const actions = {
+    single: {
+        pending: FETCH_MONUMENT_PENDING,
+        success: FETCH_MONUMENT_SUCCESS,
+        error: FETCH_MONUMENT_ERROR
+    },
+    nearby: {
+        pending: FETCH_NEARBY_MONUMENTS_PENDING,
+        success: FETCH_NEARBY_MONUMENTS_SUCCESS,
+        error: FETCH_NEARBY_MONUMENTS_ERROR
+    },
+    related: {
+        pending: FETCH_RELATED_MONUMENTS_PENDING,
+        success: FETCH_RELATED_MONUMENTS_SUCCESS,
+        error: FETCH_RELATED_MONUMENTS_ERROR
+    }
+};
```

Comment:
We should add the uris to these action objects

Reply:
I didn't follow this pattern for the ones that use template strings, it would get too confusing (``/api/monument/${id}?cascade=true``)